docs/doc_builder.py

```diff
@@ -1443,6 +1443,8 @@ def run(project_path, temp_directory, *doc_paths):
     api_path = os.path.join(base_path, 'API')
     lvgl_src_path = os.path.join(project_path, 'src')
 
+    print(f"{os.path.basename(__file__)}:  Generating API documentation .RST files...")
```

Comment:
as a suggestion I would go about handling these kinds of statements like so...


```python

def doc_print(*args):
    args = ' '.join(repr(arg) for arg in args)
    print(f'{os.path.basename(__file__)}:', args)  
```

and you use it like so...

```python
doc_print('Generating API documentation .RST files...')
```

Reply:
Now one of the things you have to remember is the gen_json script which dumps the JSON data to STDOUT That script uses the documentation build system to work. so if you have all kinds of output happening it may cause an issue. If you notice the error output is guarded by a constant. This is done so the output is able to be turned off so it doesn't pollute the STDOUT when using the gen_json script.

Reply:
Hi, Kevin!

> as a suggestion...

Thank you!  Nice logic.  I may throw that in later.

> Now one of the things you have to remember is the gen_json script which dumps the JSON data to STDOUT That script uses the documentation build system to work. so if you have all kinds of output happening it may cause an issue. If you notice the error output is guarded by a constant. This is done so the output is able to be turned off so it doesn't pollute the STDOUT when using the gen_json script.

Q1:  Having a look at this, I see that `./scripts/gen_json/pycparser_monkeypatch.py` is the only `.py` file in the `lvgl/lvgl` repository that mentions `doc_builder` (outside of `build.py`) and it calls `doc_builder.XMLSearch()`, which does not call the `doc_builder.py::run()` function, which is the only place I put `print()` calls.  So my modification does not appear to get called at any point from `gen_json.py`.  Would you concur?

Q2:  When does `gen_json.py` get used?  (I see it in `./tests/gen_json/test_gen_json.py`, but I am unclear on if that is the only place, or if it gets run as part of a test suite or by hand when _____ ?)

Reply:
I didn't go and check specifically if it would cause an issue or not. I knew that the gen_json script imported the doc_builder script. I was telling you so that you knew to be careful because it could break other things by adding print statements if they happen to end up in a location that would print from the json_gen script.


The gen_json script is used in the MicroPython Binding and I know there was work to use it in one other binding as well.

Reply:
It does one other really important feature. It collects all the documentation from LVGL and makes it easy to access for say a binding's help.

The script has also been used to locate things that need to be changed in the API. It's very handy and can be used for a plethora of different things.

Reply:
> I was telling you so that you knew to be careful because it could break other things by adding print statements if they happen to end up in a location that would print from the json_gen script.

Got it!  I was not aware that `gen_json.py` also used it, but am now, and after examining it closely, find that my `print()` calls are not at risk of interfering with `gen_json.py`.

I've observed what it does -- extracting content out of the Doxygen XML and generating an in-RAM data structure from it.  Agreed:  powerful!  :-)